content/docs/rfcs/51/README.md

```diff
@@ -173,45 +169,80 @@ This example node is part of shards `13`, `14`, and `45` in the Status main-net
 
 # Automatic Sharding
 
-> *Note:* Automatic sharding is not yet part of this specification.
-This section merely serves as an outlook.
-A specification of automatic sharding will be added to this document in a future version.
+Autosharding select shards automatically and is the default behaviour for shard choice but opting out is always possible.
+Shards (pubsub topics) not longer have to be chosen, they are instead computed from content topics with the procedure below.
+
+## Rendezvous Hashing
+
+Also known as the Highest Random Weight (HRW) method, has many properties useful for shard selection.
+- Distributed agreement. Local only computation without communication.
+- Load balancing. Content topics are spread as equally as possible to shards.
+- Minimal disruption. Content topics switch shards infrequently.
+- Low overhead. Easy to compute.
+
+### Algorithm
+
+For each shard,
+hash using Sha2-256 the concatenation of
+the content topic `application` field (N UTF-8 bytes),
+`version` (N UTF-8 bytes),
+the cluster index (2 bytes) and
+the shard index (2 bytes)
+take the first 64 bits of the hash,
+divided by 2^64,
+compute the natural logarithm of this number then
+take the negative of the weight (default 1.0) divided by it.
+Finally, sort the values and pick the shard with the highest one.
+
+## Content Topic Prefixes
+So that apps can manipulate the shard selection, 2 prefixes CAN be added to content topics.
+Generation & bias.
+When omitted default values are used.
+
+- Short format `/application/version/subject/encoding`
+- Long format `/generation/bias/application/version/subject/encoding`
+
+### Topic design
+Content topics have 2 purposes filtering and routing.
+Filtering is done by changing the `subject` field as this part is not hashed, it will not affect routing (shard selection).
+The `application` and `version` fields do affect routing.
+Application designer should understand that using different `application` field has a cost.
+It increase the traffic a node has to relay when subscribed to all topics but can benefit nodes that only use a subset of topics.
```

Comment:
Is this strictly true? In other words, if an application uses multiple application fields the impact is that the traffic for that application will be routed in different shards which adds routing complexity and _may_ increase relayed traffic. However, the application could still e.g. use filter to subscribe to content from all these shards without relaying, but this adds complexity.

I think from this RFC's perspective there is no _definitive_ knowledge about implied relay (or not) of different shards - this happens at another layer - but we could add that applications SHOULD consider the added complexity and possible bandwidth implication of possibly splitting their traffic across several shards by using multiple `application` fields. In fact, the recommendation is that application designers SHOULD group traffic that needs to be processed together with the same `application` field and SHOULD use different `application` fields for traffic that can be processed separately, to increase scalability.

Reply:
> I think from this RFC's perspective there is no definitive knowledge about implied relay (or not) of different shards

True and it also apply to this below.

> application designers SHOULD group traffic that needs to be processed together with the same application field and SHOULD use different application fields for traffic that can be processed separately, to increase scalability.

We can't make that statement. It's a case by case basis.